Adding recipe for uv-secure

owenlamont opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/uv-secure/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/uv-secure/meta.yaml**:

 * ❌ The home item is expected in the about section.
 * ❌ Non noarch packages should have python requirement without any version constraints.
 * ❌ Non noarch packages should have python requirement without any version constraints.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13087946697](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13087946697). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/uv-secure/meta.yaml```) and found it was in an excellent condition.

owenlamont commented:
@conda-forge/help-python, ready for review

ocefpaf commented on recipes/uv-secure/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+  sha256: 0aeeb2e27ebf9f5d1fef7050a1948055308cc48a0567a89972f2212b9dfcc2ba
+
+build:
+  skip: true  # [py<39]
```
This one can probably be noarch.

owenlamont replied:
Good call, updated

ocefpaf commented on recipes/uv-secure/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+    - inflect >=7.4.0
+    - pydantic >=2.10.3
+    - rich >=13.9.4
+    - tomli  # [py<311]
```
You can add a comment to drop this when updating the min Python to 3.11, but for now it is better to install tomli in later Pythons so we can have this as noarch.

```suggestion
    - tomli
```

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/uv-secure/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/uv-secure/meta.yaml**:

 * ❌ noarch: python recipes are required to have a lower bound on the python version. Typically this means putting `python >=3.6` in **both** `host` and `run` but you should check upstream for the package's Python compatibility.

For **recipes/uv-secure/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of recipe, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13214767400](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13214767400). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/uv-secure/meta.yaml```) and found it was in an excellent condition.

owenlamont commented on recipes/uv-secure/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+    - rich >=13.9.4
+    - tomli  # [py<311]
+    - typer >=0.15.1
+    - python-tzdata >=2025.1  # [win]
```
I updated the upstream repo to no longer need tzdata and bumped the version to 0.8.1 in this PR

owenlamont commented:
Hi @ocefpaf - I think I addressed all your comments when you have time to look again. Getting this win_64 CI build error but I don't think it's anything to do with my recipe... CI failed to get micromamba.exe for some reason...

```shell
Starting: Build recipes
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.250.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents: shell
call .scripts\run_win_build.bat
========================== Starting Command Output ===========================
"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\beacde62-a83b-4d3f-8c2a-c47cd5f7a361.cmd""
Provisioning build tools
Downloading micromamba 1.5.10-0
if not exist "C:\Users\VSSADM~1\AppData\Local\Temp\micromamba-16063" mkdir "C:\Users\VSSADM~1\AppData\Local\Temp\micromamba-16063"
certutil -urlcache -split -f "[https://github.com/mamba-org/micromamba-releases/releases/download/1.5.10-0/micromamba-win-64"](https://github.com/mamba-org/micromamba-releases/releases/download/1.5.10-0/micromamba-win-64%22) "C:\Users\VSSADM~1\AppData\Local\Temp\micromamba-16063\micromamba.exe"
****  Online  ****


CertUtil: -URLCache command completed successfully.
Creating environment
'"C:\Users\VSSADM~1\AppData\Local\Temp\micromamba-16063\micromamba.exe"' is not recognized as an internal or external command,
operable program or batch file.
##[error]Cmd.exe exited with code '1'.
Finishing: Build recipes
```

owenlamont commented:
Re-ran CI with a dummy commit, previous failure does appear to be a transient issue.

ocefpaf commented on recipes/uv-secure/meta.yaml:
```diff
@@ -0,0 +1,56 @@
+{% set name = "uv-secure" %}
+{% set python_min = "3.9" %}
```
You only need this if your min python is higher than conda-forge's min.

```suggestion
```

owenlamont commented:
@conda-forge-admin, please rerender
